### PR TITLE
Generate physics and navmeshes as scene-wide import options

### PR DESCRIPTION
**Pull request by SaracenOne**
This PR makes a small change to the advanced import options to all you to set an option to generate physics and navmeshes as the default option *all* mesh instances in the scene. By the default, individual nodes will follow these options, but their physics and navmesh generation options can still be overwritten on a per-node granular basis. This is consistent with existing behaviour for generating shadow meshes and LODs, and should improve workflow for importing certain dense scene with a lot of mesh instances.
godot windows tools 64_I1Bcc3B7VE

### CODE REVIEW

**Comment by jcostello** [comment]
Very nice. When doing it global, does it only generate a single body?

**Comment by SaracenOne** [comment]
> 
> 
> Very nice. When doing it global, does it only generate a single body?

Nope, it will a generate physics for every node as if each node had been setup to generate physics as as long as they're using the 'default' setting. The three generate physics a mesh instance can now have is default (copy whatever the scene wide settings are), enabled (choose custom collider settings), or disabled (never generate anything for this node)

**Comment by jcostello** [comment]
I think in some cases could be useful if you have a single object composed of several pieces but they behave as one

File: editor/import/resource_importer_scene.cpp [hunk 1]
```diff
@@ -180,6 +180,17 @@ bool ResourceImporterScene::get_option_visibility(const String &p_option, const
 		return false;
 	}
 
+	// Global settings
```
Comment (fire) [inline editor/import/resource_importer_scene.cpp:183]: Global like global to Godot Engine?

Comment (SaracenOne) [reply editor/import/resource_importer_scene.cpp:183]: Okay, I agree, these are poorly explained comments. I'll update this with better clarity.

**Comment by SaracenOne** [comment]
Updated to fix merge conflicts

**Comment by reduz** [comment]
I feel it may be better to somehow have a multi node edit setting for all these options, I am just wondering how it could work..

**Comment by SaracenOne** [comment]
@reduz My main workflow concern with not having a default collider generation setting though is the iterative approach to level design. If an artist is frequently modifying the scene geometry, it seems to make sense to me that you would want to assume that everything they add to the scene may have collision unless explicitly stated otherwise. While multi-node editing would be a nice addition, I don't think it solves the underlying problem of forcing them to manually assign collision every time they make a change, subsequently taking them out of the zone and introducing unneeded busywork.

**Comment by SaracenOne** [comment]
Rebase

**Comment by darrylryan** [comment]
@SaracenOne @reduz 
Personally I think this should neither be an import option, nor a multiselect option.
A much cleaner way would be to have a system of inheritance (think Windows NTFS permissions).

It should be possible to set flags on every node which specify that the children will have automatic colliders, LODs (could also do it for cast shadows, be used in GI etc.) The engine should then take care of this transparently, so that as nodes are added to the tree it autocreates the LODs and colliders in a background thread and caches them. If the scene is reimported it need not regenerate the LODs or colliders unless the particular mesh instance has changed. You can continue to work in the editor while the LODs and colliders are being generated and have a little toast indicator to show you they're still processing so you don't get confused why nothing is colliding yet.
The parent node's setting should appear as a greyed-out checkbox in the node itself if checked on a parent but if checked on the node itself it overrides the setting.

**Comment by SaracenOne** [comment]
I actually did consider inheritance originally, the idea the nodes above it would propogate their settings downwards to any children with the default setting, I just backed off it because I felt it was a bit over-engineered, but thinking about it some more, it might work well from a workflow perspective. Artists could essentially group different objects as children of different nodes all with their own default settings

**Comment by darrylryan** [comment]
> Artists could essentially group different objects as children of different nodes all with their own default settings

Exactly - you could have a node called "level_geometry" for example, and then all the artist has to do is import their GLTF files and nest them under that, and they get all the colliders and LODs automatically... and then if you reimport the scene it doesn't waste time rebuilding colliders and LODs unless each mesh has changed. Similarly the same method could be used to enable shadows, occluders etc. across all the meshes grouped under that node without having to go through and filter all the nodes of the same type and wait ages for the editor to single-threadedly iterate over all the nodes in e.g. a 3000 object multi-select or click them one at a time by hand.

**Comment by reduz** [comment]
Discussed in PR meeting:

I would do this as it works with tangents are other settings. Have a global setting in Meshes (not where its proposed in the PR), and then the option per mesh should be an override (that uses global default but lets you override). I think in practice this should be the easiest way to set it for all meshes and eventually disable it for one or two.

**Comment by reduz** [comment]
I am not super convinced by the implementation. IMO this should be far simpler. I would do the following changes, which are simpler:

On the Scene settings, add two enums:
`meshes/generate/collisions[Disabled | Static]`
`meshes/generate/navigation [Disabled | Enable]`
On the Mesh Instance settings:
'generate/physics [Default,Disabled,Static,Dynamic,Area]`
'generate/navigation [Default,Disabled,Mesh + NavMesh,NavMesh Only]`

Note I also merged physics options and renamed navmesh to navigation for clarity as cleanups.

**Comment by SaracenOne** [comment]
Closing this since it was pointed out to be somewhat overkill design-wise. Supporting the inclusion of this more simplified implementation instead #70256

**Pull request closed without merging**